### DISCUSSION TITLE
Why i can't use bluetooth scanner sample in samples repository?

### DISCUSSION THREAD

**Discussion by sajjadarashhh** [nanoframework/Home#1266] [Q&A]
when i wana to use that with ESP32_BLE_REV3 firmware version i dont see device names and device names is empty how can i solve that problem?

**Comment by alberk8** [comment]
Are you sure that you have a Revision 3  ESP32 hardware?.   How about try to load ESP32_BLE_REV0 firmware?

**Comment by sajjadarashhh** [comment]
@alberk8  But project run and i found devices and id's just localName is Empty
Thanks to your answer i try again with rev0 to test it!

**Comment by sajjadarashhh** [comment]
@alberk8 with that REV0  is still not show me

**Reply by alberk8** [reply]
Are you using the Bluetooth Sample code or your own code.      Try this sample [code](https://github.com/nanoframework/Samples/tree/main/samples/Bluetooth/Central1)

**Reply by sajjadarashhh** [reply]
This is bad news about this!!!
I used that sample and dont write myself...

**Reply by alberk8** [reply]
It could be your module antenna is not good.  I have tested on my module and it works.

**Reply by stefanomerotta** [reply]
Did you use an ESP32 with the UFL connector like this one?

![](https://i0.wp.com/randomnerdtutorials.com/wp-content/uploads/2020/03/ESP32-CAM-External-Antenna-connected.jpg?fit=1280%2C720&quality=100&strip=all&ssl=1)

In that case you need at least a pigtail in order to find other devices with wifi/bluetooth

**Reply by sajjadarashhh** [reply]
no i just use onboard bluetooth ESP32 WROOM Wifi Bluetooth
